src/Microsoft.IdentityModel.Tokens/TokenValidationParameters.cs

```diff
@@ -540,6 +540,11 @@ public string RoleClaimType
         /// </remarks>
         public SignatureValidatorUsingConfiguration SignatureValidatorUsingConfiguration { get; set; }
 
+        /// <summary>
+        /// Gets or sets the time provider.
+        /// </summary>
+        internal TimeProvider TimeProvider { get; set; } = TimeProvider.System;
```

Comment:
do we need to do this in more places for consistency? just curious how it could relate to this issue: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2572

Reply:
I have updated the places where we used it with ValidationParameters.

I believe that if we make this change public, it will address #2572.
For now, I am only using it in my tests and all other scenarios behave the same as without it.